pkg/apis/serving/v1beta1/revision_defaults.go

```diff
@@ -18,7 +18,49 @@ package v1beta1
 
 import (
 	"context"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/knative/serving/pkg/apis/config"
 )
 
+// SetDefaults implements apis.Defaultable
 func (r *Revision) SetDefaults(ctx context.Context) {
+	r.Spec.SetDefaults(ctx)
+}
+
+// SetDefaults implements apis.Defaultable
+func (rts *RevisionTemplateSpec) SetDefaults(ctx context.Context) {
+	rts.Spec.SetDefaults(ctx)
+}
+
+// SetDefaults implements apis.Defaultable
+func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
+	cfg := config.FromContextOrDefaults(ctx)
+
+	// Default TimeoutSeconds based on our configmap
+	if rs.TimeoutSeconds == nil {
+		ts := cfg.Defaults.RevisionTimeoutSeconds
+		rs.TimeoutSeconds = &ts
+	}
+
+	var container corev1.Container
+	if len(rs.PodSpec.Containers) == 1 {
+		container = rs.PodSpec.Containers[0]
+	}
+	defer func() {
+		rs.PodSpec.Containers = []corev1.Container{container}
+	}()
+
+	if container.Resources.Requests == nil {
```

Comment:
Check Resources for nil?

Reply:
Resources is just an inline struct, Requests is a map, which will panic if we don't initialize it before checking for CPU.